pandas/_libs/parsers.pyx

```diff
@@ -607,8 +607,8 @@ cdef class TextReader:
             void *ptr
 
         if not hasattr(source, "read"):
-            raise IOError(f'Expected file path name or file-like object, '
-                          f'got {type(source)} type')
+            raise TypeError('Expected file path name or file-like object, '
```

Comment:
can you construct a test that actually hits this? (e.g. you didn't have to change any tests here)

Reply:
Added `pandas/tests/io/parser/common/test_common_basic.py::test_read_filepath_or_buffer` which checks four parsers.